src/github/api.rs

```diff
@@ -54,6 +57,12 @@ pub trait Session: Send + Sync {
         comment: Option<&str>,
     ) -> Result<()>;
     fn get_timeline(&self, owner: &str, repo: &str, number: u32) -> Result<Vec<TimelineEvent>>;
+
+    // checks api
+    fn get_suites(&self, pr: &PullRequest) -> Result<Vec<CheckSuite>>;
+    fn get_check_run(&self, pr: &PullRequest, id: u32) -> Result<CheckRun>;
+    fn create_check_run(&self, pr: &PullRequest, run: &CheckRun) -> Result<u32>;
```

Comment:
This is the only API call that this PR really needs/uses. Added the rest of the Checks API here (from pr-kicker) for completeness.